transport/rif_comms/node.py

```diff
@@ -1,135 +1,228 @@
+                self.log.info("incoming message", message=raiden_message)
+                self._handle_message(raiden_message)
+
+    @staticmethod
+    def _notification_to_message(notification_data: ChannelNewData) -> RaidenMessage:
+        """
+        :param notification_data: raw data received by the RIF Comms GRPC API
+        :return: a raiden.Message
+        """
+        content_text = notification_data.data
+        """
+        ChannelNewData has the following structure:
+            from: "16Uiu2HAm8wq7GpkmTDqBxb4eKGfa2Yos79DabTgSXXF4PcHaDhWJ"
+            data: "{\"type\":\"Buffer\",\"data\":[104,101,121]}"
+            nonce: "\216f\225\232d\023e{"
+            channel {
+              channelId: "16Uiu2HAm9otWzXBcFm7WC2Qufp2h1mpRxK1oox289omHTcKgrpRA"
+            }
+        """
+        if content_text:
+            # we first transform the content of the notification data to a dictionary
+            content = json.loads(content_text.decode())
+            # the message is inside the notification data, encoded by the RIF Comms GRPC api
+            message_string = bytes(content["data"]).decode()
+            message_dict = json.loads(message_string)
+            return message_from_dict(message_dict)
+        return None
+
+    def _handle_message(self, message: RaidenMessage):
+        """
+        Handle received Raiden message.
+        """
+        if self.stop_event.ready():
+            return  # ignore when node is stopped
+
+        # process message if its type is expected
+        if isinstance(message, (Delivered, Processed, SignedRetrieableMessage)):
+            self.log.info(
+                "Raiden message received",
+                type=type(message),
+                node=pex(self._raiden_service.address),
+                message=message,
+                sender=pex(message.sender),
+            )
 
-        return f"<{self.__class__.__name__}{node} id:{id(self)}>"
+            try:
+                # acknowledge to sender that their message was received
+                self._ack_message(message)
+            except (InvalidAddress, UnknownAddress, UnknownTokenAddress):
+                self.log.warning("exception while processing message", exc_info=True)
+                return
+
+            # once acknowledged, pass message to raiden service for business logic
+            self._raiden_service.on_message(message)
+
+        else:
+            self.log.warning(
+                "unexpected type of message received",
+                type=type(message),
+                node=pex(self._raiden_service.address),
+                message=message,
+            )
+
+    def _ack_message(self, message: (Delivered, Processed, SignedRetrieableMessage)):
+        """
+        Acknowledge a received Raiden message by sending a Delivered-type message back.
+        If the received Raiden message is of the Delivered type, no action is taken.
+        """
+        if not isinstance(message, Delivered):
```

Comment:
ok, so we call _ack_message from delivered but skip the response here..

im ok with this

Reply:
lets move this to handle_message